### PR TITLE
WIP: Test out new rubocop config

### PR DESCRIPTION
**Pull request by alexjfisher**
This is a test of https://github.com/voxpupuli/voxpupuli-test/pull/40

### CODE REVIEW

**Comment by alexjfisher** [comment]
This PR is just an example of what sort of changes we would have to make if we upgrade rubocop.
DO NOT MERGE.

File: lib/puppet/functions/extlib/has_module.rb [hunk 1]
```diff
@@ -28,6 +30,7 @@ def has_module(module_name) # rubocop:disable Style/PredicateName
 
     metadata = JSON.parse(File.read(metadata_json))
     return true if metadata['name'] == full_module_name
+
     false
```
Comment (ekohl) [inline lib/puppet/functions/extlib/has_module.rb:34]: This is even prettier:
```suggestion
    metadata['name'] == full_module_name
```

File: spec/functions/extlib/cache_data_spec.rb [hunk 2]
```diff
@@ -14,14 +16,14 @@
   describe 'data caching' do
     context 'when not in cache' do
       it 'returns supplied data' do
-        is_expected.to run.with_params(namespace, data_name, initial_data).and_return(initial_data)
+        expect(subject).to run.with_params(namespace, data_name, initial_data).and_return(initial_data)
```
Comment (ekohl) [inline spec/functions/extlib/cache_data_spec.rb:19]: This is new and surprises me. I think this is https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecimplicitexpect but not 100% sure. Why show a difference in line 10/11 (which also appears to be the exact same test)?

File: spec/functions/extlib/dump_args_spec.rb [hunk 3]
```diff
@@ -1,30 +1,32 @@
...
   end
 
   context 'when called with a hash' do
     let(:hash) { { 'param_one' => 'value', 'param_two' => 42 } }
 
     it 'puts pretty JSON' do
-      allow(STDOUT).to receive(:puts).with(JSON.pretty_generate(hash))
-      is_expected.to run.with_params(hash)
+      allow($stdout).to receive(:puts).with(JSON.pretty_generate(hash))
+      expect(subject).to run.with_params(hash)
```
Comment (ekohl) [inline spec/functions/extlib/dump_args_spec.rb:20]: I think I saw a cleaner way for this in https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecexpectoutput but not sure how we should apply that here.

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -1,18 +1,18 @@
+# frozen_string_literal: true
```
Comment (ekohl) [inline spec/spec_helper.rb:1]: This should be fixed in our modulesync.

**Review by ekohl — COMMENTED** [review]
I'm a bit concerned about the `is_expected.to` change. That's the dominant way we write our specs and could be a big change. Other than that it looks ok to me.

File: spec/spec_helper.rb (hunk 4 above)
Comment (alexjfisher) [reply spec/spec_helper.rb:1]: yeah, this was just a quick demo PR.  I spent no real time verifying the autocorrected changes. Some were generated with `rubocop -a` and others with `rubocop -A`. I should have really split each cop into a separate commit to make it more obvious.

File: lib/puppet/functions/extlib/has_module.rb (hunk 1 above)
Comment (ekohl) [reply lib/puppet/functions/extlib/has_module.rb:34]: https://github.com/voxpupuli/puppet-extlib/pull/178

**Pull request closed without merging**